handlers/main.yml

```diff
@@ -95,11 +95,11 @@
     nohup bash -c "
     returncode=0
     {% for interface in all_interfaces_changed | reverse %}
-    ifdown {{ interface }};
+    ifdown --allow auto {{ interface }};
```

Comment:
This won't work on RHEL-derived systems - there ifdown/ifup only takes one argument, the interface name. Could you add a short comment to explain what this is for? I understand myself now but had to read a few man pages first :)

Reply:
Hmm, I see.

On debian system the `--allow` flag to ifup/ifdown behaves like a filter to the interfaces configured in `/etc/network/interfaces`. Typically you'd configure an interface there like `auto eth0`, meaning the state of the interface is managed by ifup/ifdown. 

Not so for hot-pluggable wifi interfaces (e.g. usb wifi dongle). Here you would configure them like `allow-hotplug wlan0`. This means the interface's state is managed by the kernel's hotplug events. So if you plug the wifi dongle in, the interface will be brought up and configured (static IP or dhcp or whatever). If you unplug the wifi dongle, the interface will be brought down. Trying to manage a hot-pluggable interface using ifup/ifdown will just fail.

Long story short: On Debian you cannot bounce an interface marked with `allow-hotplug` using ifup/ifdown. That's why I added the `--allow auto` argument to ifup/ifdown. This will only be effective for interface marked with `auto`. Interface marked with `allow-hotplug` will simply be ignored then.

On a debian system also see:
```
$ man interfaces
$ man ifup
```

@markgoddard I can implement a conditional switch here for RHEL systems. Or do you see another way?

Reply:
Thanks, seems like a reasonable solution to the problem. How about this (untested):

`ifup {% if ansible_os_family == 'Debian' %}--allow auto {% endif %}{{ interface }};`

Reply:
Also, could you put a brief explanation as a comment in the code?

Reply:
@markgoddard Looks good. Give me some time for the change and to test. Might only be at the weekend.

Btw, do you see any possibility to do a negative test for RHEL on Travis. Travis is only using Ubuntu after all. :-/

Reply:
It would be possible by using a RHEL docker container. It would be a bit fiddly though.